### PR TITLE
Add support for taking named tunnel credentials from an environment var

### LINKED ISSUE #457: Feature Request: Allow providing named tunnel credentials via environment variables.
I'm currently looking to run `cloudflared` within AWS Fargate, providing access to other AWS resources within my VPC via Cloudflare Tunnel.

Unfortunately, as mentioned in #230, there's not really a clean way to provide the required credentials to run a named tunnel to the container as we don't have the ability to mount files into the container easily.

In my mind the best way to solve this problem is to allow passing the contents of the credentials JSON file as an environment variable, so it can easily be populated from AWS Secrets Manager / AWS SSM Parameter Store.

### PR DESCRIPTION
**Pull request by nint8835**
# Description
Resolves #457.

Adds a new flag, `credentials-contents`, to act as an alternative to `credentials-file`. This flag has an environment variable of `TUNNEL_CRED_CONTENTS` attached, allowing providing of named tunnel credentials via an environment variable. This flag, if provided, causes `subcommandContext.findCredentials` to load the credentials from the value of the flag, rather than trying to find a file on disk to load from.

# Testing
- I have added unit tests for the behaviour I implemented
- I have tested locally by running the following:
```bash
make cloudflared
export TUNNEL_CRED_CONTENTS=$(cat ~/.cloudflared/TUNNEL_UUID.json)
./cloudflared tunnel run --url http://localhost:5000 TUNNEL_UUID
``` 
- I have successfully gotten `cloudflared` running on Fargate in AWS ECS, and ensured it is functioning properly

### CODE REVIEW

File: cmd/cloudflared/tunnel/subcommands.go
Comment (nmldiegues) [inline cmd/cloudflared/tunnel/subcommands.go:118]: Consider adding here some description that relates this with `credentials-file`:  if both are set, then `credentials-contents` takes precedence.

File: cmd/cloudflared/tunnel/subcommand_context_test.go
Comment (nmldiegues) [inline cmd/cloudflared/tunnel/subcommand_context_test.go:195]: s/file/contents/

File: cmd/cloudflared/tunnel/subcommand_context_test.go
Comment (nmldiegues) [inline cmd/cloudflared/tunnel/subcommand_context_test.go:217]: same as above

**Review by nmldiegues — COMMENTED** [review]
Thanks for this contribution.

I only have minor comments, otherwise LGTM

**Review by nmldiegues — APPROVED** [review]

**Review by sssilver — CHANGES_REQUESTED** [review]
Can we squash before getting this merged?

**Review by sssilver — APPROVED** [review]

**Pull request merged**
